corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -70,6 +82,156 @@ hqDefine("geospatial/js/case_grouping_map",[
+                paint: {
+                    'circle-color': 'red',
+                    'circle-radius': 10,
+                    'circle-stroke-width': 1,
+                    'circle-stroke-color': '#fff',
+                },
+            });
+        });
+        mapboxInstance.on('moveend', updateClusterStats);
+
+        return mapboxInstance;
+    }
+
+    function updateClusterStats() {
+        const sourceFeatures = map.querySourceFeatures('caseWithGPS', {
+            filter: ['get', 'cluster'],
+        });
+
+        // Mapbox clustering creates the same cluster groups with slightly different coordinates.
+        // Seems to be related to keeping track of clusters at different zoom levels.
+        // There could therefore be more than one cluster that share the same ID so we should keep track
+        // of these to skip them if we've gone over them already
+        let uniqueClusterIds = {};
+        let clusterStats = {
+            total: 0,
+            min: 0,
+            max: 0,
+        };
+        for (const clusterFeature of sourceFeatures) {
+            // Skip over duplicate clusters
+            if (uniqueClusterIds[clusterFeature.id]) {
+                continue;
+            }
+
+            uniqueClusterIds[clusterFeature.id] = true;
+            clusterStats.total++;
+            const pointCount = clusterFeature.properties.point_count;
+            if (pointCount < clusterStats.min || clusterStats.min === 0) {
+                clusterStats.min = pointCount;
+            }
+            if (pointCount > clusterStats.max) {
+                clusterStats.max = pointCount;
+            }
+        }
+        clusterStatsInstance.totalClusters(clusterStats.total);
+        clusterStatsInstance.clusterMinCount(clusterStats.min);
+        clusterStatsInstance.clusterMaxCount(clusterStats.max);
+    }
+
+    function loadMapClusters(caseList) {
+        let caseLocationsGeoJson = {
+            "type": "FeatureCollection",
+            "features": [],
+        };
+
+        _.each(caseList, function (caseWithGPS) {
+            const coordinates = caseWithGPS.coordinates;
+            if (coordinates && coordinates.lat && coordinates.lng) {
+                caseLocationsGeoJson["features"].push(
+                    {
+                        "type": "feature",
+                        "properties": {
+                            "id": caseWithGPS.caseId,
+                        },
+                        "geometry": {
+                            "type": "Point",
+                            "coordinates": [coordinates.lng, coordinates.lat],
+                        },
+                    }
+                );
+            }
+        });
+
+        if (map.getSource('caseWithGPS')) {
+            map.getSource('caseWithGPS').setData(caseLocationsGeoJson);
+        } else {
+            map.on('load', () => {
+                map.getSource('caseWithGPS').setData(caseLocationsGeoJson);
+            });
+        }
```

Comment:
@zandre-eng 
Why can't we just do `map.on('load', ...)` without the if...else...? Or is the loadMapClusters also triggered when zooming?

Reply:
@Charl1996 If the user switches to a different page, then the map should already be initialized and `if (map.getSource()` should resolve to `true` as the map source is already loaded. This might not be the case when the user refreshes the page, and the clusters might get loaded before the map is done initializing. In this scenario, `if (map.getSource()` will be `false` and we offload pushing the data to when the `load` event is called.